nemo/core/classes/modelPT.py

```diff
@@ -1735,6 +1744,75 @@ def update_save_restore_connector(cls, save_restore_connector):
+            if self.cfg.chakra_profile.get('enabled', False):
+                self._chakra_profile_enabled = True
+                self._chakra_profile_start_step = self.cfg.chakra_profile.get('start_step', 0)
+                self._chakra_profile_end_step = self.cfg.chakra_profile.get('end_step', 0)
+
+                if self.cfg.chakra_profile.get('trace_dir', None) is None:
+                    raise ValueError(
+                        f'chakra profile output path ({self._memory_profile_output_path}) is not set.'
+                    )
+
+                trace_dir = Path(self.cfg.chakra_profile.get('trace_dir'))
+                warmup_steps = self.cfg.chakra_profile.get('warmup_steps', 0)
+                active_steps = self.cfg.chakra_profile.get('active_steps', 1)
+
+                from nemo.utils.env_var_parsing import get_envint
+                job_id = get_envint("SLURM_JOB_ID", 0)
+
+                self._chakra_trace_dir = trace_dir / f'{job_id}_chakra'
+                self._kineto_trace_dir = trace_dir / f'{job_id}_kineto'
+
+                self._chakra_trace_dir.mkdir(parents=True, exist_ok=True)
+                self._kineto_trace_dir.mkdir(parents=True, exist_ok=True)
+
+                if type(self._chakra_profile_start_step) == int:
+                    logging.info(f'chakra profiling setup with start_step: {self._chakra_profile_start_step}')
+                else:
+                    raise ValueError(
+                        f'chakra start_step must be of type int. Found: {type(self._chakra_profile_start_step)}'
+                    )
+
+                if type(self._chakra_profile_end_step) == int:
+                    logging.info(f'chakra profiling setup with end_step: {self._chakra_profile_end_step}')
+                else:
+                    raise ValueError(f'chakra end_step must be of type int. Found: {type(self._chakra_profile_end_step)}')
+
+                if self._chakra_profile_end_step >= self._chakra_profile_start_step:
+                    pass
+                else:
+                    raise ValueError(f'chakra end_step must be greater than or equal to chakra start_step')
+
+                if self.cfg.nsys_profile.get('enabled', False):
+                    logging.warning(f'Profiling with Nsys enabled will result in the Kineto trace losing GPU performance data.')
+
+                self._et = ExecutionTraceObserver()
+                self._prof = torch.profiler.profile(
+                    activities=[torch.profiler.ProfilerActivity.CPU,torch.profiler.ProfilerActivity.CUDA,],
+                    schedule=torch.profiler.schedule(
+                        wait=0,
+                        warmup=warmup_steps,
+                        active=active_steps
+                    ),
+                    record_shapes=True,
+                    execution_trace_observer=self._et,
+                    experimental_config=_ExperimentalConfig(enable_cuda_sync_events=True)
```

Comment:
Why we need `record_shapes` and `experimental_config` here? Are them necessary?

Reply:
I removed these 2 options in the latest commit